libgpos/src/task/CAutoTaskProxy.cpp

```diff
@@ -506,18 +506,14 @@ CAutoTaskProxy::Execute
 	{
 		// mark task as erroneous
 		ptsk->SetStatus(CTask::EtsError);
-
-		if (m_fPropagateError)
-		{
-			GPOS_RETHROW(ex);
-		}
+		GPOS_RETHROW(ex);
```

Comment:
m_fPropageError tells should I copy the exception from sub task to current task. But in this particular function ((i.e) CAutoTaskProxy::Execute) no multiple thread is spawned (Because it call its own worker to execute this task).
Only CAutoTaskProxy::Schedule function will spawn multiple thread.

You are right that we removed the code path that eat the exception. But does it make sense if we run task in the same thread as parent & we want to catch all exception and pretend as if nothing happened ?
If I am spawning new thread and then it need to copy the exception to parent to propagate. CAutoTaskProxy does it by default and may make sense in some case to disable that through m_fPropageSubTaskError. 


Which line in _api.cpp you referred for changing false to true ?